src/main/java/digital/slovensko/autogram/core/eforms/UpvsEFormResources.java

```diff
@@ -73,4 +74,16 @@ public boolean findResources() throws XMLValidationException {
 
         return true;
     }
+
+    private byte[] getRemoteResource(String url) {
+        var bytes = getResource(url);
+        if (bytes == null)
+            return null;
+
+        var s = new String(bytes, ENCODING);
+        if (s.contains("<!DOCTYPE html>") || (s.contains("<html>") && s.contains("<title>Nedostupnosť portálu</title>")))
```

Comment:
Stale sa mi zda toto fragilne. Ovela radsej by som nejako spravil to, ze budeme matchovat voci spravnemu obsahu - toto su vsetko xslt transformacie alebo nevieme tam nic taketo najst?

Reply:
obsah môže byť ale všeličo. XSLT, XSD, manifest.xml. To by sme pri každom requeste museli parsovať XML response a potom validovať voči 3 XSD. Pri UPVS formulári ide vždy minimálne o 3 stiahnuté súbory, ale môže to by aj viac. Čiže 9x validácia XSD - to by už bolo imho cítiť.

Toto riešenie je trochu fragilné, ale je rýchle a prakticky úplne stačí.

Reply:
Ide proste o to, že v tomto prípade treba trochu opustiť keďže tento problém sa nedá opraviť až tak pekne. Jasné, šlo by sa s tým pohrať, ale pravdepodobne by to bolo pomalšie, a ako už aj Marek povedal, bolo by to trochu na tom cítiť. Hlavne, ak opravia tú chybu na slovensko.sk, čo verím, že áno, tak ide momentálne iba o krátkodobé riešenie.

Reply:
A nestaci teda zistovat, ze to vracia xml (trebars hlavicka) a nie html?

Reply:
Môžeme testovať teda pre `<?xml version="1.0" encoding="utf-8"?>`, keďže každý xml to obsahuje (niektoré majú encoding veľkým, UTF-8, ale to môžeme vyriešiť jednoducho tým, že to dáme do lower case. Čo myslíš? Podľa mňa je to v podstate jedno, `<!DOCTYPE html>` sa určite nikde nenachádza (minimálne by sa teda nemal)

Reply:
Skor som mal na mysli http hlavicky

Reply:
tie súbory ťaháme cez nejaký FileCacheDataLoader z DSS, aby za nás riešil cachovanie. IMHO sa k hlavičkám nedostaneme.

Reply:
Presne tak, k tomuto sa nemáme ako dostať ak by sme nenaimplementovali nejaký vlastný `CommonsDataLoader` a pár vecí z `FileCacheDataLoader`, čo bolo viac menej len duplikácia kódu. Mohli by sme ešte requestovať 2x (u nás a potom DSS samotné), čo sa tiež ale neoplatí.